src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/TilkjentYtelseUtils.kt

```diff
@@ -61,13 +62,23 @@ object TilkjentYtelseUtils {
 
         return innvilgedePeriodeResultaterBarna.flatMap { periodeResultatBarn ->
             relevanteSøkerPerioder.filter { it.overlapper(periodeResultatBarn) }
-                .map { overlappendePeriodeResultatSøker ->
+                .mapNotNull { overlappendePeriodeResultatSøker ->
                     val barn = barnaIdenter[periodeResultatBarn.aktør.aktørId] ?: throw Feil("Finner ikke barn")
                     // beregn beløpsperiode med sats og tilsvarende gjeledende prosentandel
                     val beløpsperiode = beregnBeløpsperiode(
                         overlappendePeriodeResultatSøker,
                         periodeResultatBarn
                     )
+                    // beløpsperiode kan ikke starte etter barnet fyller 2 år
+                    val maksTomDatoIMellom1Og2ÅrVilkår = periodeResultatBarn.vilkårResultater
+                        .filter { it.vilkårType == Vilkår.MELLOM_1_OG_2_ELLER_ADOPTERT && it.periodeTom != null }
+                        .maxBy { checkNotNull(it.periodeTom) }.periodeTom
+                    if (beløpsperiode.fom.toLocalDate().isAfter(maksTomDatoIMellom1Og2ÅrVilkår)) {
+                        return@mapNotNull null
+                    }
```

Comment:
Hva skjer dersom `maksTomDatoIMellom1og2ÅrVilkår` er `null`. Tror ikke vi har noe validering i frontend som sier at tom må settes på `MELLOM_1_OG_2_ELLER_ADOPTERT` vilkåret 🤔 Men det burde vi kanskje hatt?

Reply:
Dersom `tom` ikke er satt tror jeg du kan anta at `tom` er `fom + 1år`.

Reply:
@bragejahren Settes til TIDENES_ENDE for så vidt. Det finnes allerede en validering i Behandlingsresultat som stopper behandling hvis tom ikke er lik barnets 2 års dato.